packages/lesswrong/components/messaging/NewConversationButton.jsx

```diff
@@ -6,54 +6,32 @@ Button used to start a new conversation for a given user
 
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import Dialog from 'material-ui/Dialog';
-import { Components, registerComponent, withCurrentUser, getFragment } from 'meteor/vulcan:core';
+import { Components, registerComponent, withCurrentUser, withNew } from 'meteor/vulcan:core';
 import {  withRouter } from 'react-router';
 import Conversations from '../../lib/collections/conversations/collection.js';
 
 class NewConversationButton extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {
-      open: false,
-    }
-  }
-  handleOpen = () => {
-    this.setState({open: true});
-  }
-  handleClose = () => {
-    this.setState({open: false});
+
+  newConversation = () => {
+    const { user, currentUser, newMutation, router } = this.props
+
+    newMutation({
+      collection: Conversations,
+      document: {participantIds:[user._id, currentUser._id]},
+      currentUser: currentUser,
+      validate: false,
+    }).then(response=>{
```

Comment:
I would transform this into an `await` instead of a `then`, and make the whole `newConversation` function async